woocommerce/woocommerce
# Ensure subscriptions menu (experiment) is only visible to eligible stores

### All Submissions:






### Changes proposed in this Pull Request:

This PR prevents rendering the **WooCommerce > Subscriptions** menu item for WooCommerce stores that are ineligible for the experiment, as defined in #32682.

A store is **ineligible** for the experiment if:
- The store has WooCommerce Payments installed.
- The store has any subscription plugins installed, as discussed here: pdjTHR-WQ-p2.
- The store address is not USA (eligible for [WCPay Subscriptions](https://woocommerce.com/document/payments/subscriptions/)).
- The store has:
  - Been live for < 6 months.
  - Had < 1 sale in last 30 days.

Conflicting subscription plugins, installed and confirmed to ensure ineligibility:
- [x] [WooCommerce Payments](https://en-au.wordpress.org/plugins/woocommerce-payments/) (.org) 
- [x] [YITH WooCommerce Subscription](https://wordpress.org/plugins/yith-woocommerce-subscription/) (.org)
- [x] [Subscriptions for WooCommerce](https://en-au.wordpress.org/plugins/subscriptions-for-woocommerce/) (.org)
- [ ] [Subscriptions for WooCommerce Pro](https://wpswings.com/product/subscriptions-for-woocommerce-pro/) (WP Swings)
   - Plugin slug: `woocommerce-subscriptions-pro` (confirmed via Jetpack Site Profiles)
- [x] [WooCommerce Subscriptions](https://woocommerce.com/products/woocommerce-subscriptions/) (WooCommerce.com)
- [x] [SUMO Subscriptions – WooCommerce Subscription System](https://codecanyon.net/item/sumo-subscriptions-woocommerce-subscription-system/16486054) (Envato)
- [x] [Subscriptio](https://codecanyon.net/item/subscriptio-woocommerce-subscriptions/8754068) (Envato)
- [ ] [Subscriptions for WooCommerce](https://www.webtoffee.com/product/woocommerce-subscriptions/) (Webtoffee)
   - Plugin slug: `xa-woocommerce-subscriptions` (confirmed via Jetpack Site Profiles)


Part of #32682

### How to test the changes in this Pull Request:

1. Checkout this branch. I recommend running a fresh `wp-env` from the `plugins/woocommerce` directory, so you can delete plugins in an isolated environment.
2. Make your store eligible for the experiment:
	1. **Delete** any subscription plugins and WooCommerce Payments (if installed).
	2. Set your store’s address to a location in USA.
	3. Set your store’s installation timestamp to a time >= 6 months age. Do this by updating the `woocommerce_admin_install_timestamp` option in the `wp_options` table to a Unix timestamp that is >= 6 months in the past.  
	I installed the [WP Console](https://wordpress.org/plugins/wp-console/) plugin and used it to run the following:
	   `update_option( 'woocommerce_admin_install_timestamp', strtotime( '-6 months' ) );`
	5. If your store has not had a completed order in the last 30 days, create a test order and mark it as completed.
	   Delete the existing cached transient: Using [WP Console](https://wordpress.org/plugins/wp-console/), run `delete_transient( 'woocommerce-wcpay-subscriptions_recent_sales_eligibility' );`
3. You should see the **WooCommerce > Subscriptions** menu item and be able to click on it to view the empty state page.
5. **Test store location ineligibility**:
	1. Set your store’s address to a location in a country other than USA.
	2. Refresh/navigate to WooCommerce admin, the **WooCommerce > Subscriptions** menu item should **not** be visible.
	3. Reset the store location to USA to become eligible, as per step 2.
6. **Test store age ineligibility**:
	1. Set your store’s installation timestamp to a time < 6 months age, using method from step 2.
	   `update_option( 'woocommerce_admin_install_timestamp', strtotime( '-1 month' ) );`
	3. Refresh/navigate to WooCommerce admin, the **WooCommerce > Subscriptions** menu item should **not** be visible.
	4. Reset the store age to become eligible, as per step 2.
7. **Store sales ineligibility**:
	1. Delete any orders from the last 30 days.
	2. Delete the existing transient cache:
	   Using [WP Console](https://wordpress.org/plugins/wp-console/), run `delete_transient( 'woocommerce-wcpay-subscriptions_recent_sales_eligibility' );`
	4. Refresh/navigate to WooCommerce admin, the **WooCommerce > Subscriptions** menu item should **not** be visible.
	5. Create a test order and set the order status to **Completed**, then delete the cached transient to become eligible, as per step 2.
8. **WooCommerce Payments ineligibility**:
	1. Install and optionally activate WooCommerce Payments.
	2. Refresh/navigate to WooCommerce admin, the **WooCommerce > Subscriptions** menu item **should be visible**. Clicking on this should navigate you to the WCPay Subscriptions page, rather than the empty state experiment page.
	3. Delete the plugin to become eligible, as per step 2.
9. **Subscriptions plugin ineligibility** (repeat the following steps for multiple subscriptions plugins):
	1. Install and optionally activate a subscription plugin.
	2. Refresh/navigate to WooCommerce admin.
		1. If a **WooCommerce > Subscriptions** menu item exists, clicking it should not navigate to the empty state subscriptions page.
	3. Delete the plugin to become eligible, as per step 2.

### Other information:




### FOR PR REVIEWER ONLY:
